haystack/retriever/dense.py

```diff
@@ -552,54 +522,210 @@ def retrieve(self, query: str, filters: dict = None, top_k: Optional[int] = None
             top_k = self.top_k
         if index is None:
             index = self.document_store.index
-        query_emb = self.embed(texts=[query])
+        query_emb = self.embed_queries(texts=[query])
         documents = self.document_store.query_by_embedding(query_emb=query_emb[0], filters=filters,
                                                            top_k=top_k, index=index)
         return documents
 
-    def embed(self, texts: Union[List[List[str]], List[str], str]) -> List[np.ndarray]:
+    def embed_queries(self, texts: List[str]) -> List[np.ndarray]:
         """
-        Create embeddings for each text in a list of texts using the retrievers model (`self.embedding_model`)
+        Create embeddings for a list of queries.
 
-        :param texts: Texts to embed
-        :return: List of embeddings (one per input text). Each embedding is a list of floats.
+        :param texts: Queries to embed
+        :return: Embeddings, one per input queries
         """
-
         # for backward compatibility: cast pure str input
         if isinstance(texts, str):
             texts = [texts]
         assert isinstance(texts, list), "Expecting a list of texts, i.e. create_embeddings(texts=['text1',...])"
+        return self.embedding_encoder.embed_queries(texts)
+
+    def embed_passages(self, docs: List[Document]) -> List[np.ndarray]:
+        """
+        Create embeddings for a list of passages.
+
+        :param docs: List of documents to embed
+        :return: Embeddings, one per input passage
+        """
+        return self.embedding_encoder.embed_passages(docs)
 
-        if self.model_format == "farm" or self.model_format == "transformers":
-            # TODO: FARM's `sample_to_features_text` need to fix following warning -
-            # tokenization_utils.py:460: FutureWarning: `is_pretokenized` is deprecated and will be removed in a future version, use `is_split_into_words` instead.
-            emb = self.embedding_model.inference_from_dicts(dicts=[{"text": t} for t in texts])
-            emb = [(r["vec"]) for r in emb]
-        elif self.model_format == "sentence_transformers":
-            # texts can be a list of strings or a list of [title, text]
-            # get back list of numpy embedding vectors
-            emb = self.embedding_model.encode(texts, batch_size=200, show_progress_bar=False)
-            emb = [r for r in emb]
-        return emb
 
+class _EmbeddingEncoder:
```

Comment:
I am in favour to move all `EmbeddingEncoder` related code to another file. This file is becoming very big also very hard to navigate around code.

Reply:
I don't mind tbh, except IIRC we wanted to keep everything in dense.py including EmbeddingRetriever. Perhaps we can leave this for a future refactor @tholor ?

Reply:
Yes, let's leave this for a future refactor. I can see that we need to rethink the module structure here in a more holistic way.